### DISCUSSION TITLE
Logs are not created for all logging levels except ERROR (Unlocked package)

### DISCUSSION THREAD

**Discussion by dAzzl3** [jongpie/NebulaLogger#481] [Q&A] [answered]
Hi All,
I'm not sure I understood how this framework works. The problem I've faced is that if I log anything with logging level ERROR, Logs are created, but if I want to log something with INFO, DEBUG ent. Logs are not created.
So this code will create a log
`Logger.error('Add log entry using Nebula Logger with logging level == ERROR');
Logger.saveLog();`
but this one won't
`Logger.debug('Add log entry using Nebula Logger with logging level == DEBUG');
Logger.saveLog();`

This happen only in Prod. And my user is assigned `Logger: Admin` permission set.

Does anybody knows how handle this? It will be much appreciated ;)

**Comment by jongpie** [comment]
Hi @dAzzl3 - this sounds like you have `LoggerSettings__c` record(s) configured in your production org that are configured to only save certain logging levels. You can check this by either opening the custom tab "Logger Settings", or by going to Setup --> Custom Settings --> Logger Settings. In particular, you'll want to see what value(s) are set for the "Logging Level" field.

**Reply by dAzzl3** [reply]
Hi @jongpie, thanks for fast reply.
I have checked that and there is no records of `LoggerSettings__c` in Prod nor in Sandboxes. =(

I have run the a script to see which Logging Level are available except ERROR.

```
Logger.error('Add log entry using Nebula Logger with logging level == ERROR');

Logger.warn('Add log entry using Nebula Logger with logging level == WARN');

Logger.info('Add log entry using Nebula Logger with logging level == INFO');

Logger.debug('Add log entry using Nebula Logger with logging level == DEBUG');

Logger.fine('Add log entry using Nebula Logger with logging level == FINE');

Logger.finer('Add log entry using Nebula Logger with logging level == FINER');

Logger.finest('Add log entry using Nebula Logger with logging level == FINEST');
Logger.saveLog();
```

This time only `INFO, WARN, ERROR` Log entries were created. But I would like to use `DEBUG` in my code, is there any setting I need to change?

**Reply by jongpie** [reply, marked as answer]
Hi @dAzzl3 - in some versions of Nebula Logger, by default, only ERROR, WARN, and INFO are used if you don't have any `LoggerSettings__c` records created. 

You'll need to create a `LoggerSettings__c` record for the organization, with a logging level of DEBUG, FINE, FINER, or FINEST (depending on which levels you want to be enabled).
